test/e2e/replica_set_mongod_port_change_with_arbiters/replica_set_mongod_port_change_with_arbiters_test.go

```diff
@@ -0,0 +1,79 @@
+package replica_set_mongod_config
+
+import (
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/mongodb/mongodb-kubernetes-operator/pkg/automationconfig"
+
+	. "github.com/mongodb/mongodb-kubernetes-operator/test/e2e/util/mongotester"
+
+	e2eutil "github.com/mongodb/mongodb-kubernetes-operator/test/e2e"
+	"github.com/mongodb/mongodb-kubernetes-operator/test/e2e/mongodbtests"
+	setup "github.com/mongodb/mongodb-kubernetes-operator/test/e2e/setup"
+)
+
+func TestMain(m *testing.M) {
+	code, err := e2eutil.RunTest(m)
+	if err != nil {
+		fmt.Println(err)
+	}
+	os.Exit(code)
+}
+
+func TestReplicaSetMongodPortChangeWithArbiters(t *testing.T) {
+	ctx := setup.Setup(t)
+	defer ctx.Teardown()
+
+	mdb, user := e2eutil.NewTestMongoDB(ctx, "mdb0", "")
+	scramUser := mdb.GetScramUsers()[0]
+
+	_, err := setup.GeneratePasswordForUser(ctx, user, "")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	tester, err := FromResource(t, mdb)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	connectivityTests := func(t *testing.T) {
+		fmt.Printf("connectionStringForUser: %s\n", mongodbtests.GetConnectionStringForUser(mdb, scramUser))
+		t.Run("Test Basic Connectivity with generated connection string secret",
+			tester.ConnectivitySucceeds(WithURI(mongodbtests.GetConnectionStringForUser(mdb, scramUser))))
+
+		// FIXME after port change in the service mongodb+srv connection stopped working!
```

Comment:
I have confirmed that this is most probably an issue with e2e tests/mongodb go driver only. I've confirmed that SRV query after port change is returning the new port correctly. Also mongosh connection to the replicaset using mongodb-srv connection string from secret is working correctly.